gravitee-am-repository/gravitee-am-repository-jdbc/src/main/java/io/gravitee/am/repository/jdbc/management/api/JdbcCredentialRepository.java

```diff
@@ -65,6 +65,13 @@ public Flowable<Credential> findByUsername(ReferenceType referenceType, String r
                 .map(this::toEntity);
     }
 
+    @Override
+    public Flowable<Credential> findByUsername(ReferenceType referenceType, String referenceId, String username, int latestLimit) {
+        LOGGER.debug("findByUsername({},{},{},{})", referenceType, referenceId, username, latestLimit);
+        return credentialRepository.findByUsernameWithLatestLimit(referenceType.name(), referenceId, username, latestLimit)
+                .map(this::toEntity);
+    }
+
```

Comment:
Question: shouldn't we remove the oldest credential when the new one is above the limit?

Reply:
Not yet. It would be a feature request for that. For now, as a bug fix, it's the only necessity.